pandas/tests/indexes/datetimes/test_arithmetic.py

```diff
@@ -933,3 +935,16 @@ def test_datetime64_with_DateOffset(klass, assert_func):
                  Timestamp('2000-02-29', tz='US/Central')], name='a')
     assert_func(result, exp)
     assert_func(result2, exp)
+
+
+@pytest.mark.parametrize('years,months', product([-1, 0, 1], [-2, 0, 2]))
```

Comment:
If you parametrize years & months separately, pytest will handle the `product` 

i.e. `@pytest.mark.parametrize(('years',[-1, 0, 1]), ('months', [-2, 0, 2]))`